app/models/amendment.rb

```diff
@@ -3,6 +3,7 @@
 class Amendment < ApplicationRecord
   belongs_to :user
   belongs_to :budget
+  validate :type_can_not_change, on: :update
```

Comment:
You can use a conditional validation here (and remove the call to `locked_type?` in line 29):

```ruby
validate :type_can_not_change, on: :update, if: :locked_type?
```
Also, you can call it `:type_cannot_be_changed` following the naming convention from [the guides](https://guides.rubyonrails.org/active_record_validations.html#custom-methods).

Reply:
And solved too